Automattic/jetpack
# Slideshow Shortcode: Adding the Jetpack namespace to CSS class names

Fixes #17715

#### Changes proposed in this Pull Request:

* Changes proposed in this PR include changes to CSS classes within the [slideshow-shortcode.css ](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/shortcodes/css/slideshow-shortcode.css) file (prefixing them with `jetpack-`). 
* This is to prevent CSS name collisions. 
* This involves changes to the CSS file as well as the PHP and JS files where the classnames are first defined. It also involves changes to theme compatibility files which reference some of the CSS classes (for [TwentyFourteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentyfourteen.css), [TwentyFifteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentyfifteen.css) and [TwentySixteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentysixteen.css)). Lastly, it also includes a change to the [test-class-slideshow](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.slideshow.php) PHP file to include an updated CSS classname where relevant.

#### Additional comments:

* In testing the theme compat files, I noticed no change on my test site to classnames (when viewing in the inspector) either before or after changing the CSS classnames in those files. That seemed to be because all slideshows were already using the `jetpack` namespaced classnames after the change to the three main slideshow-shortcode files (css, js and php). It may be that there is another way to test slideshows that doesn’t use block classnames, however the theme compat files have been changed anyway to be safe.


#### Jetpack product discussion:

* Internal discussion here: p1643907890247239-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:






* To test, add a slideshow to a post or page using the shortcode. Here is an example:  `[gallery type="slideshow" ids="22,21,19,20"]` - however you’d need to replace those IDs with image IDs from your test site. The end result should look similar to this:

<img width="323" alt="Slideshow-via-shortcode" src="https://user-images.githubusercontent.com/16754605/152547470-ba1ed78b-9e30-4aff-b187-a57402dfa9dc.png">


* Other ways to test the shortcode version of the slideshow include adding the Classic Widgets plugin, and adding a gallery widget to a test site, making sure the gallery is a slideshow. Checking the inspector should confirm that the jetpack prefix is being added to class names (examples will include `jetpack-slideshow-window`, `jetpack-slideshow-slide`).
